purchases/src/main/kotlin/com/revenuecat/purchases/common/diagnostics/DiagnosticsSynchronizer.kt

```diff
@@ -45,10 +45,12 @@ internal class DiagnosticsSynchronizer(
     }
 
     fun clearDiagnosticsFileIfTooBig() {
-        if (diagnosticsFileHelper.isDiagnosticsFileTooBig()) {
-            verboseLog("Diagnostics file is too big. Deleting it.")
-            diagnosticsTracker.trackMaxEventsStoredLimitReached()
-            resetDiagnosticsStatus()
+        enqueue {
```

Comment:
It should be fine to enqueue since we only have a single diagnostics file, where the track methods are called. So if we have to delete the diagnostics file, it should happen before we add any more events.